HTTP body match isn't working

https://github.com/racker/rackspace-monitoring-poller/blob/master/check/check_http.go#L188

In the CEP I'm seeing `"status": "metric \"body_match\" is not available",` for a check with `details.body` set.


https://github.com/racker/rackspace-monitoring-poller/blob/master/check/check_http.go#L196
```
[Tue, 11 Apr 2017 01:12:09 UTC] DEBUG ch2UEibgmf:remote.http: Debug Request method=GET url=http://monitoring.rackspace.net:80
panic: runtime error: index out of range

goroutine 32 [running]:
github.com/racker/rackspace-monitoring-poller/check.(*HTTPCheck).Run(0x18a83440, 0x0, 0x0, 0x0)
	/home/travis/gopath/src/github.com/racker/rackspace-monitoring-poller/check/check_http.go:196 +0x2c2f
github.com/racker/rackspace-monitoring-poller/poller.(*EleScheduler).Execute(0x18944f40, 0x85c34a0, 0x18a83440)
	/home/travis/gopath/src/github.com/racker/rackspace-monitoring-poller/poller/scheduler.go:286 +0x2a7
github.com/racker/rackspace-monitoring-poller/poller.(*EleScheduler).runCheckTimerLoop(0x18944f40, 0x85c34a0, 0x18a83440)
	/home/travis/gopath/src/github.com/racker/rackspace-monitoring-poller/poller/scheduler.go:269 +0x3c4
created by github.com/racker/rackspace-monitoring-poller/poller.(*EleScheduler).Schedule
	/home/travis/gopath/src/github.com/racker/rackspace-monitoring-poller/poller/scheduler.go:251 +0x60
```

--- a/check/check.go
+++ b/check/check.go
@@ -356,13 +356,11 @@ func (ch *Base) AddTLSMetrics(cr *Result, state tls.ConnectionState) *TLSMetrics
 }
 
 func (ch *Base) readLimit(conn io.Reader, limit int64) ([]byte, error) {
-	bytes := make([]byte, limit)
-	bio := io.LimitReader(conn, limit)
-	count, err := bio.Read(bytes)
+	body, err := ioutil.ReadAll(io.LimitReader(conn, limit))
 	if err != nil && err != io.EOF {
 		return nil, err
 	}
-	return bytes[:count], nil
+	return body, nil
 }
 
 func ReadCheckFromFile(filename string) (Check, error) {

--- a/check/check_http.go
+++ b/check/check_http.go
@@ -193,9 +193,9 @@ func (ch *HTTPCheck) Run() (*ResultSet, error) {
 			return crs, nil
 		}
 		if m := re.FindSubmatch(body); m != nil {
-			cr.AddMetric(metric.NewMetric("body_match", "", metric.MetricString, string(m[1]), ""))
+			cr.AddMetric(metric.NewMetric("body_match", "", metric.MetricString, string(m[0]), "string"))
 		} else {
-			cr.AddMetric(metric.NewMetric("body_match", "", metric.MetricString, "", ""))
+			cr.AddMetric(metric.NewMetric("body_match", "", metric.MetricString, "", "string"))
 		}
 	}
 

